File: C++_Solutions/SAPO/School_Round_2017/Round_2/Problem_1_Secret_Code_Solution_1.cpp
#include <iostream>
#include <string>

using namespace std;

void replace_spaces(string& message){
  for (size_t i = 0; i < message.size(); i++) {
    if (message[i] == ' ') {
      message[i] = '.';
    }
  }
  string add = "";
  for (size_t i = 0; i < message.size()%5; i++) {
    add+='.';
  }
  message = message + add;
}

string flip(string message){
  string out = "";
  for (size_t i = 0; i < 5; i++) {
    for (size_t j = 0; j < message.size()/5; j++) {
      out += message[5*j+i];
    }
  }
  return out;
}

int main(int argc, char const *argv[]) {
  string message = "T.GLEHC.L.EACEM.THDAR.A.LATSATTHET!.EDE.T....HDKT.EOIH";
  replace_spaces(message);
  cout<<flip(message)<<endl;
  return 0;
}

File: Python Solutions/SAPO/School Round 2017/Round 2/Problem 1 Secret Code Solution 1.py
if __name__ == '__main__':
    message = input('Input: ')
    message+=' '*(len(message)%5)
    message = message.replace(' ','.')
    message=list(message)
    encrypted = ''
    vertical = len(message)//5
    for i in range(5):
        for j in range(vertical):
            index = j*5 + i
            encrypted+=message[index]
    print(encrypted)

'''
T.GLEHC.L.EACEM.THDAR.A.LATSATTHET!.EDE.T....HDKT.EOIH
'''